extensions/agent/client/src/agentUtils.ts

```diff
@@ -0,0 +1,47 @@
+'use strict';
+
+import * as sqlops from 'sqlops';
+
+export class AgentUtils {
+
+	private static _agentService: sqlops.AgentServicesProvider;
+	private static _connectionService: sqlops.ConnectionProvider;
+	private static _fileBrowserService: sqlops.FileBrowserProvider;
+	private static _queryProvider: sqlops.QueryProvider;
+
+	public static async getAgentService(): Promise<sqlops.AgentServicesProvider> {
+		if (!AgentUtils._agentService) {
+			let currentConnection = await sqlops.connection.getCurrentConnection();
+			this._agentService = sqlops.dataprotocol.getProvider<sqlops.AgentServicesProvider>(currentConnection.providerName, sqlops.DataProviderType.AgentServicesProvider);
+		}
+		return AgentUtils._agentService;
+	}
+
+	public static async getDatabases(ownerUri: string): Promise<string[]> {
+		if (!AgentUtils._connectionService) {
+			let currentConnection = await sqlops.connection.getCurrentConnection();
+			this._connectionService = sqlops.dataprotocol.getProvider<sqlops.ConnectionProvider>(currentConnection.providerName, sqlops.DataProviderType.ConnectionProvider);
+		}
+		return AgentUtils._connectionService.listDatabases(ownerUri).then(result => {
+			if (result && result.databaseNames && result.databaseNames.length > 0) {
+				return result.databaseNames;
+			}
+		});
+	}
+
+	public static async getFileBrowserService(ownerUri: string): Promise<sqlops.FileBrowserProvider> {
+		if (!AgentUtils._fileBrowserService) {
+			let currentConnection = await sqlops.connection.getCurrentConnection();
```

Comment:
same comment about null `currentConnection`?  Is it safe to assume `getCurrentConnection` will always return a connection and that the connect is the correct one?  I won't add this comment on all uses, but please consider this comment for all usages of `getCurrentConnection`.